.github/workflows/ci.yml

```diff
@@ -185,11 +185,17 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Configure Developer Command Prompt for Microsoft Visual C++
-        # Using microsoft/setup-msbuild is not enough.
-        uses: ilammy/msvc-dev-cmd@v1
-        with:
-          arch: x64
+      - name: Set up VS Developer Prompt
+        shell: pwsh
+        run: |
+          $vswherePath = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+          $installationPath = & 'C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe' -latest -property installationPath
+          if ($installationPath -and (test-path "$installationPath\Common7\Tools\vsdevcmd.bat")) {
+            & "${env:COMSPEC}" /s /c "`"$installationPath\Common7\Tools\vsdevcmd.bat`" -no_logo && set" | foreach-object {
+              $name, $value = $_ -split '=', 2
+              echo "$name=$value" >> $env:GITHUB_ENV
```

Comment:
This part seems too invasive to me. Please compare the list of environment variables in the "Get tool information" step on the master branch with those in this PR.

Reply:
> The only tool we need is msbuild.exe...

We also use now `mt.exe`: https://github.com/bitcoin/bitcoin/blob/dc403f0c13e1ceee3d4651b4056ff2bad47d0315/.github/workflows/ci.yml#L246-L253

Reply:
> Shall I just put it back?

Perhaps updating only the `Path` variable will be sufficient?

Reply:
> I will look at using the top commit hash for cache invalidation. If I've understood you correct the current cache key is:
> 
> `hashFiles('cmake_version', 'msbuild_version', 'toolset_version', 'vcpkg.json')`
> 
> we think this will be sufficient:
> 
> `hashFiles('vcpkg_commit_hash', 'vcpkg.json')`

Yes, I thin so.

> Which then means we can drop the vswhere tool finding.

Right. However, we still need to locate `mt.exe`.

Reply:
@hebasto compelling enough to ACK as is?